### PR TITLE
Generalize VectorTools::get_position_vector for simplex meshes

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Comment by peterrum** [comment]
Failing test is unrelated and is fixed in #11573.

File: include/deal.II/numerics/vector_tools_interpolate.templates.h [hunk 1]
```diff
@@ -653,8 +653,12 @@ namespace VectorTools
       {
         const Quadrature<dim> quad(fe.get_unit_support_points());
 
-        MappingQGeneric<dim, spacedim> map_q(fe.degree);
-        FEValues<dim, spacedim> fe_v(map_q, fe, quad, update_quadrature_points);
+        const auto map_q = ReferenceCell::get_default_mapping<dim, spacedim>(
+          fe.reference_cell_type(), fe.degree);
+        FEValues<dim, spacedim>              fe_v(*map_q,
```
Comment (drwells) [inline include/deal.II/numerics/vector_tools_interpolate.templates.h:658]: I'm not going to hold up the PR for it, but is it just me or is clang-format getting weirder?

**Review by drwells — APPROVED** [review]
I think this is fine in isolation but it doesn't match my recollection of what we agreed on for implementing the geometry-agnostic replacement for `MappingQ1` last Friday. Did we write that down somewhere so I can check my memory?

**Comment by drwells** [comment]
On the other hand this is consistent with #11539 so, even if we want to do something else later, we make the implementation more uniform by merging both so lets merge.

**Pull request merged**